lib/core/api/src/lib/adf-http-client.service.ts

```diff
@@ -332,12 +319,16 @@ export class AdfHttpClient implements ee.Emitter,JsApiHttpClient {
     }
 
     private createCSRFToken(a?: any): string {
-        const randomValue = window.crypto.getRandomValues(new Uint32Array(1))[0];
+        const randomValue = AdfHttpClient.getSecureRandomValue();
```

Comment:
This is the only code change in this file. The other changes seem to be related to code formatting, and were not showing up on my IDE

Reply:
Just out of curiosity what was failing in that case?

Reply:
The `window.crypto.getRandomValues()` method returns a truly random int number (i.e. a cryptographically strong/unguessable number) between 0 to whatever the max value for the passed in parameter is. Prior to ADF 6.5.0 , the older independent JS-API library used  similar method of generating a token, only instead it used `Math.random()`. This method returned a seemingly random decimal number between 0 and 1. Using `Math.random()` here, instead of `window.crypto.getRandomValue()`, also fixes the issue, but is not suggested as numbers generated via `Math.random()` do have an internal pattern which can lead to a security breach. 

What I've done here is, generate a random number using `window.crypto.getRandomValue()`, and then divide that by the max value that was posssible for this type. This makes the number to be within the range of 0 to 1, and makes the login work again. 

Why the login only works for numbers between the 0-1 range is something that I've not been able to figure out though.